### ISSUE TITLE
The Artifact isn't specified even I build a war for artifact

### ISSUE THREAD

**Issue by xmeng1** [GoogleCloudPlatform/google-cloud-intellij#1625]
(Please ensure you are running the latest version of Cloud Tools for IntelliJ with _Help > Check for Updates_.)

- Cloud Tools for IntelliJ version: 17.8
- Google Cloud SDK version _(run `gcloud version` on the command line)_: 
Google Cloud SDK 168.0.0
alpha 2017.03.24
app-engine-java 1.9.54
app-engine-python 1.9.57
beta 2017.03.24
bq 2.0.25
cloud-datastore-emulator 1.2.1
core 2017.08.21
docker-credential-gcr 
gcd-emulator v1beta3-1.0.0
gcloud 
gsutil 4.27
kubectl 
pubsub-emulator 2017.08.07
- OS: fedora 24

**What did you do?**

upgrade to the new version plugin : 17.8.2. 

**What did you expect to see?**

The run configuration should be right

**What did you see instead?**
I always got the error in the running configuration as follows, I have added build Artifact in the build list, but it cannot find the Artifact to deploy. I don't why, and is there any tool or log to see what the wrong with the project. (I also try to create a new project, the configuration is right. I think the reason is that the project is an old project which uses the old version plugin.)

![image](https://user-images.githubusercontent.com/5753142/29964943-ab8eb4e0-8f03-11e7-8a49-bd4a6011cda5.png)

By the way, I I use the version of **google-cloud-tools-17.2.6_2017**, the previous version of run configuration do not have the error. Does the version after  17.2.6 change the run configuration mechanism? because the configurations work on 17.2.6 cannot work on the version after 17.2.6. Conversely, the configurations work on the version after 17.2.6 cannot work on the version 17.2.6.

![image](https://user-images.githubusercontent.com/5753142/29965440-6d27589a-8f05-11e7-84e6-7b46982465ae.png)

(screenshots are helpful)

_Feel free to deviate from this template as needed, especially if you are submitting a feature request._

**Comment by etanshaul** [comment]
hi @xmeng1. sorry you're running into this.

Can you check to see what your Artifact settings in you Module Settings are:
![image](https://user-images.githubusercontent.com/1735744/29976436-061bf31a-8f08-11e7-9198-30a7736dadb3.png)

You can access this by right clicking on the root of your project -> Module Settings -> Artifacts.

You _should_ have an exploded war artifact present. If not, you can try creating a new one (make sure its exploded).

The new version of the plugin _shouldn't_ have affected this; so lets try and debug further to see what the problem is.

**Comment by etanshaul** [comment]
@xmeng1, another thing you could try doing in the meantime is creating a new local run/debug configuration to see if it helps:

1) Click to edit your run configs:
![image](https://user-images.githubusercontent.com/1735744/29976963-d1a463d6-8f09-11e7-8fac-a8c40c507c5d.png)

2) Click the + button to add a new "Google App Engine Standard Local Server" (it shouldn't be hidden, but if you don't see it, scroll to the button and click "irrelevant...")
![image](https://user-images.githubusercontent.com/1735744/29976988-e8020868-8f09-11e7-99fc-6f6ae154212b.png)

3) Configure the new run config. Give it a new name, select an artifact to deploy (if one shows up), and see if it behaves as expected.

**Comment by nkibler** [comment]
@xmeng1 I'm closing this issue for now since it's not reproducible. Please re-open it if you're still having this problem. Thanks!

**Comment by lilianaziolek** [comment]
@nkibler I'm having exactly same problem :/ Quite annoyingly I cannot narrow down what seems to be the exact issue, usually straight after importing a gradle project it works, and then it stops working and I have no idea why! My project structure is:
root
--java
----appengine
----someotherstuff
--othernonjavastuff

I import into intellij from gradle.build on the root level. I wonder if the more sophisticated project structure causes intellij and/or cloud plugin to get confused? I do see the artifacts (war and exploded war) when I press F4 (in artifacts section) - but they do not appear in the dropdown. As said, it seems to briefly work straight after import into intellij, but stops after a couple of builds...
I tried adding the Appengine Facet on both root project and appengine project and neither seems to help... Interestingly, the Deploy task seems to be able to see the artifact just fine.
Any way I can help you further to reproduce it ?

**Comment by nkibler** [comment]
@lilianaziolek thanks for the report! I'll dig deeper into this and try to find out what's going on. In the meantime, could you screenshot and post your project's list of artifacts, as @etanshaul did above in https://github.com/GoogleCloudPlatform/google-cloud-intellij/issues/1625#issuecomment-326609949?

**Comment by etanshaul** [comment]
@lilianaziolek in addition to what @nkibler requested, could you also please post a screenshot of your module page (also under 'Project Settings').

And just to confirm: you are getting this issue only in the App Engine local development run configuration, but not from the App Engine deployment configuration?

**Comment by lilianaziolek** [comment]
So right now it works (I literally just reimported the project from scratch, after deleting .idea folder).
Screenshots below. The absolutely maddening thing is that these three screens look exactly the same way as it did half an hour ago when it was broken (obviously except for the fact that artifact dropdown did not have entries). I'll keep an eye on it for when it breaks again and see if I can find a pattern or any differences in intellij folder - so far can't reliably reproduce it so I know it's going to be tricky to find! Sorry can't be more helpful.
As for Deploy vs Run - yes, it did seem to see the artifact in the deploy config. But not sure if that means it always works, or was just a coincidence, I usually deploy from cmd line so it's possible this also has the issue - I just didn't come across it
![artifacts](https://user-images.githubusercontent.com/271914/32955807-f1ea1500-cbae-11e7-8f40-4783b9154206.PNG)
![project](https://user-images.githubusercontent.com/271914/32955809-f20474cc-cbae-11e7-93fb-8fab97ed7c29.PNG)
![run](https://user-images.githubusercontent.com/271914/32955810-f21bc00a-cbae-11e7-95cc-31870e14b8b8.PNG)


.

**Comment by etanshaul** [comment]
@lilianaziolek this is very strange. Next time it happens, could you try doing a gradle sync to see if that solves it:

<img width="411" alt="screen shot 2017-11-17 at 10 53 20 am" src="https://user-images.githubusercontent.com/1735744/32955971-8abe6116-cb85-11e7-9512-61c3f7d2ea9f.png">

(Click on the refresh icon in the gradle panel)

**Comment by lilianaziolek** [comment]
@etanshaul I think I did try that, as well as forcing a project build, forcing artifacts build, running cmd line gradle build, restarting intellij and whatever other "refresh" options I could think of. Nothing short of shutting down intellij, deleting .idea and re-importing seemed to help. But I'm sure there is a reason, just haven't found it yet! :)
I'll update here if I have any more insights. For now my gut feeling is it has something to do with my multi-module gradle setup. Maybe at some point I changed some project settings and intellij/plugin gets confused. I don't recall having this issue back when my project was an independent maven project just pulling in the dependencies as jars.

**Comment by lilianaziolek** [comment]
Also, I just backed up the whole project as a zip (including .idea and .iml files, and even build directory). If it breaks again, I'll try running a diff to see if there are any hints.

**Comment by freddyboucher** [comment]
Exact same issue:

- Google Cloud SDK 179.0.0
- app-engine-java 1.9.59
- app-engine-python 1.9.62
- bq 2.0.27
- core 2017.11.06
- gcloud 
- gsutil 4.28

IntelliJ Google Cloud Tools
- Version: 17.11.1

<img width="1178" alt="run debug configurations" src="https://user-images.githubusercontent.com/695120/33164326-548f0f2c-d086-11e7-8ac2-4769a2e18338.png">

<img width="1203" alt="modules" src="https://user-images.githubusercontent.com/695120/33164329-59539eba-d086-11e7-8d34-ff060edb2ee2.png">

<img width="1199" alt="facets" src="https://user-images.githubusercontent.com/695120/33164338-62c84a18-d086-11e7-9b46-4ba6e5f9bb2f.png">

<img width="1199" alt="artifacts" src="https://user-images.githubusercontent.com/695120/33164343-68afb524-d086-11e7-84d2-5a6b655cb67c.png">

**Comment by etanshaul** [comment]
@freddyboucher we are looking into this issue. See the latest comments [here](https://github.com/GoogleCloudPlatform/google-cloud-intellij/issues/1758) and a possible workaround using the maven or gradle plugins.

**Comment by lilianaziolek** [comment]
So, my update is: I've lost my option to run local again, and I've done literally no changes in my project's config, checked these few windows that I pasted screenshots of earlier, and they look identical (with the exception of artifact missing from deployment dropdown). More interestingly, because it was already created, despite it showing a red cross on it, I can actually start it, and it DOES work... but I can't create a new configuration as there is no artifact to select. It looks like for some reason at some point the plugin stops seeing the artifact even though it is there...

**Comment by etanshaul** [comment]
@xmeng1 @lilianaziolek @freddyboucher 

For those of you still experiencing this issue, could you please try the following:

It is possible that you either don't have App Engine framework support at all, or it has been added to the incorrect module. App Engine framework support is required for an artifact to be available for running on the local app engine development server. To attempt to resolve this, add the support via the tools menu (**make sure you are on plugin version 17.11.1 or later**):

![image](https://user-images.githubusercontent.com/1735744/33290226-114ad716-d390-11e7-8999-219e73b99889.png)
And then select the module to which you'd like to add the support.

This may resolve the problem and result in the `Artifact to Deploy` dropdown being populated again:
![image](https://user-images.githubusercontent.com/1735744/33291379-9a90a2be-d393-11e7-83b5-21a4c6da4ad0.png) 

Also, if you could please report back on which build system you are using (Gradle / Maven / none ..) that would be helpful too in providing a full detailed explanation and a resolution to this.

We have been able to reproduce certain instances of this issue specifically with a multi-module project using Gradle. The issue being that the IDE does not prevent you from adding App Engine framework support on a newly imported project before the Gradle subsystem has finished resolving all of the modules (causing the Framework Support to be added to the wrong module).

**Comment by freddyboucher** [comment]
@etanshaul 
I'm using Maven.
Here what I've exactly done: https://youtu.be/LlASfj22PuQ

**Comment by etanshaul** [comment]
@freddyboucher thank you for making this!

I haven’t inspected it too closely yet (I will), but one thing that immediately jumped out at me is that you deleted the artifact manually. Any particular reason? You can try to recreat the artifact for the particular module you are interest in running locally (make sure it’s an exploded war since you are on App Engine standard).

**Comment by freddyboucher** [comment]
@etanshaul 
I deleted artifacts just to start clean.
IntelliJ re-creates them automatically.

**Comment by freddyboucher** [comment]
@etanshaul 
So reimporting the whole project fixed the issue for me:
- delete App Engine Facet
- delete Artifacts
- delete App Engine Run/Configuration
- delete .idea folder
- close project
- reopen project
- Reimport All Maven projects
- Fix Project SDK
- Run Google App Engine Local Server

https://youtu.be/8nLbF86R8Yg

**Comment by lilianaziolek** [comment]
Hi @etanshaul - I am using a multi-module gradle project, so that would be in-line with what you say. Also, as said, didn't have that issue when I had a stand-alone maven project for appengine - again, suggesting something with the multi-modularity might be to blame. 
In my case the very weird thing is that it doesn't happen immediately on import - it breaks further down the line, after I've been using the project for a while, and recreating project actually helps. I have the appengine support added to both root and child (appengine) project - can you confirm if it should be attached to only one of them (which?) or to both? I will certainly try removing and re-adding the facet when it breaks (although I could swear I tried that already...).

**Comment by etanshaul** [comment]
@freddyboucher thanks for the update

@lilianaziolek - regarding which module the app engine support should be added to will depend on your specific project. Typically, you will have a root (parent) module that houses one or more child modules which are the actual "deployable" artifacts. 

Using this [multi-module guestbook](https://github.com/GoogleCloudPlatform/appengine-modules-sample-java) app as an example, the modules you would be interested in are the child `appengine-modules-guestbook`, and `appengine-modules-shardedcounter` modules. The following is the shows the state of the Google App Engine framework support that is automatically configured by the plugin:
<img width="863" alt="screen shot 2017-11-28 at 1 14 24 pm" src="https://user-images.githubusercontent.com/1735744/33336505-25feb930-d43e-11e7-92d3-dbd370f743cd.png">

**Comment by eschultink** [comment]
@etanshaul I'm getting this same  issue with Intellij plugin version 17.12.2.   My project is multi-module java8, built with Maven using the [Google Cloud Tools-based plugin](https://github.com/GoogleCloudPlatform/app-maven-plugin), version 1.3.2. 

The example you cite ([multi-module guestbook](https://github.com/GoogleCloudPlatform/appengine-modules-sample-java)) is java7 with an EAR structure. EAR is NOT supported for Java8 according to [configuration files documentation](https://cloud.google.com/appengine/docs/standard/java/configuration-files), yet Java7 is deprecated.  Is there an up-to-data example to look at that works with the IntelliJ plugin?

Outside of IntelliJ, I am able to run the GAE dev server using the Maven plugin directly from the root of my GAE `default` module (using `mvn appengine:run`).  But just like the others on this thread, in IntelliJ, the artifacts drop-down for the GAE Run Configuration is never populated with any of the exploded wars in my project.

From inside Intellij, I can run the dev server using a Maven run configuration with the `appengine:run` goal - but no changes I make to files inside IntelliJ are reflected unless I kill and restart that process every time. 

My project tree looks as follows:
<img width="691" alt="intellij-project-structure" src="https://user-images.githubusercontent.com/102831/34015488-6f7aff5a-e0d3-11e7-9f4c-871476958aac.png">

The Google Cloud Tools-based Maven plugin is used only in each of the GAE service poms  (red above).  Each of these uses WAR packing.

The other Maven modules (green) get built into JARs, which are then dependencies for the GAE service modules.

**Comment by eschultink** [comment]
@etanshaul BTW, the suggested trick of using **Tools** --> **Google Cloud Tools** --> **Add App Engine** --> **Google App Engine Standard** does not work BUT does nuke my existing appengine-web.xml file, leaving only:
```xml
<?xml version="1.0" encoding="utf-8"?>
<appengine-web-app xmlns="http://appengine.google.com/ns/1.0">
  <threadsafe>true</threadsafe>
  <runtime>java8</runtime>
</appengine-web-app>
```
I had a bunch of stuff in there before, although some of it I think is no longer used in java8 - one important want was `<instance-class>`, which I think is still needed?

**Comment by etanshaul** [comment]
Hi @eschultink, you are right that EAR is not supported for java 8. That example was just meant to show the running of its war sub-modules individually which should work for java 7 or 8. 

I'll see if I can find or create a working example.

Just to confirm, they are missing from the local dev server run configuration, but are they present in the App Engine deployment configuration?
![image](https://user-images.githubusercontent.com/1735744/34017004-ebc028c4-e0f1-11e7-8d38-be836ba9937b.png)

Thanks for the details, we'll work to get to the bottom this.

**Comment by etanshaul** [comment]
Yeah, the version of Java should not affect whether or not you need `<instance-class>`. 

Also, the fact that you were able to add Google App Engine standard support to your module via that tools menu shortcut suggests to me that something was misconfigured previously with your App Engine framework support (which controls the App Engine facet). If you already had this facet in place, the tools menu would not have presented you with the option to add it again.

Its tough to say whats going on without seeing screenshots of your module / facet screen under `Project Structure -> Modules / Facets`.

**Comment by chanseokoh** [comment]
> From inside Intellij, I can run the dev server using a Maven run configuration with the appengine:run goal - but no changes I make to files inside IntelliJ are reflected unless I kill and restart that process every time.

`appengine.fullscan.seconds=1` may work: https://github.com/GoogleCloudPlatform/app-maven-plugin/blob/master/USER_GUIDE.md#how-do-i-enable-hot-reload-of-my-application

I don't know about the IntelliJ plugin, but if it directly updates classes at the location where the dev appserver of the Maven plugin launched, the hot reload might work.

**Comment by lilianaziolek** [comment]
So in the last week I had the run-local option go red-crossed (no artifact) and back to working at least once (could have been more, I work on a few other modules in same project so don't always notice immediately).
Is there perhaps a chance of you creating a build with more logging to help you troubleshoot it, and I'd be happy to use it for some time? Would it help shed more light at what makes it click and what breaks it?

**Comment by etanshaul** [comment]
@lilianaziolek good idea. We're taking another close look at this. I'll keep you updated.

**Comment by eschultink** [comment]
@chanseokoh thanks, that certainly improves the situation.  With that, I'm able to leave GAE dev server running (having started it with `mvn appengine:run`), then just do `mvn war:exploded` as you suggest to get the dev server to reflect changes to JS/HTML/etc in ~5-10 seconds. It's better than restarting the whole dev server.

I think if IntelliJ can be setup correctly, editing files in IntelliJ would trigger something similar behind the scenes - as IntelliJ would understand that file `example.js` is used to build the war:exploded artifact, so rebuild/update the artifact when the file is edited.

And yes - hot reload of JARs compiled from the *java* files would be ideal; but JS/HTML alone is an improvement.

**Comment by eschultink** [comment]
@etanshaul  
> Also, the fact that you were able to add Google App Engine standard support to your module via that tools menu shortcut suggests to me that something was misconfigured previously with your App Engine framework support

Sorry - I wasn't clear; I think manually deleted the facet from the Maven modules before trying that. Otherwise, it didn't include my *actual* GAE modules in the list of modules I could add GAE too.

But anyways, I found success doing the following:

1. close the project; delete it from IntelliJ; delete ALL the IntelliJ files: both `.idea` directory **AND** all `.iml` files throughout the project hierarchy (`find . -name '*.iml' -delete`). I think the `.iml` was what I missed before. 

2. re-import to IntelliJ, trying to keep it as plain-vanilla with the project settings as possible. Really the only things I think I changed from defaults:
   * Maven 3.5 from my machine, instead of the copy bundled in intellij
   * Java8 SDK instead of 7

This gave me the following:

![project - artifacts](https://user-images.githubusercontent.com/102831/34022340-8bbb32be-e0f3-11e7-9f2a-eff189192142.png)
![project - facest](https://user-images.githubusercontent.com/102831/34022341-8bd4a730-e0f3-11e7-9e30-fe9c37bb8615.png)
![project - modules](https://user-images.githubusercontent.com/102831/34022342-8bef225e-e0f3-11e7-9ed2-54bd16ef4d2b.png)

That was unusual. Before (trying with existing project, and previous import attempts), the GAE facets were always added to all the modules; they were just wrong as above. 

But also in the Event Log I had a message:
```
5:10 PM	Frameworks Detected
				Google App Engine Standard, Python frameworks are detected.
				Configure
```

Clicking "Configure", it detected the 3 GAE modules. I just followed the next couple steps, and then proper a proper Run Configuration with an artifact value was added.  Curiously, it picked the wrong artifact (not my default module, which has the `<services>` defined in its `pom.xml`); and it only found 2 of my 3 artifacts for the list.  But the other one was the default, so I was able to start up the dev server properly:

![run configs](https://user-images.githubusercontent.com/102831/34022641-3b7e6904-e0f5-11e7-9a76-f63ca77b4952.png)

**Comment by etanshaul** [comment]
@eschultink, thanks - these details are very helpful.

- When you say you had `<services>` defined in your pom, did you mean to say `<modules>`?
- Regarding your 'default' module: this is not an App Engine module in the sense that it has an `appengine-web.xml` correct? The plugin will currently only pick up those modules that have an `appengine-web.xml` as targets for deployment.
- With the current state of the local server run configuration, are you missing any functionality that you had when running the server using the `app-maven-plugin`? We have an open issue (https://github.com/GoogleCloudPlatform/google-cloud-intellij/issues/1235) which would enhance the local server run configuration to allow selection of maven modules as deployment targets which should allow runs to be more maven-aware.

**Comment by eschultink** [comment]
@etanshaul 
latest is that this morning, having been doing a few changes locally, switching branches, etc - I've lost my `module-frontend:war exploded` artifact from the Run Configuration dropdown again (similar to how a few others have said that it happens randomly). Curiously, I still see another GAE module in the list - but not the one that is my "default" service, that's the main web entrypoint. 

> When you say you had `<services>` defined in your pom, did you mean to say `<modules>`?

I mean that it has the following defined, which is what [GAE Maven plugin expects](https://cloud.google.com/appengine/docs/standard/java/tools/migrate-mavenhttps://cloud.google.com/appengine/docs/standard/java/tools/migrate-maven) to see in the "default" Maven module of a multi-service GAE project in java8 standard environment. 
```xml
            <plugin>
                <groupId>com.google.cloud.tools</groupId>
                <artifactId>appengine-maven-plugin</artifactId>
                <version>${maven.appengine.plugin-version}</version>
                <configuration>
                    <project>${appengine.app.projectid}</project>
                    <version>${module.version}</version>
                    <services>
                        <service>${targetDirectory}</service>
                        <service>${project.parent.basedir}/module-other/target/module-other-${project.version}</service>
                        <service>${project.parent.basedir}/module-another/target/module-another-${project.version}</service>
                    </services>
                </configuration>
            </plugin>
```

>Regarding your 'default' module: this is not an App Engine module in the sense that it has an `appengine-web.xml` correct? The plugin will currently only pick up those modules that have an `appengine-web.xml` as targets for deployment.

it is a Maven *module* to be deployed as my `default` App Engine *service*. (what GAE now calls "services" used to be called "modules", as I understand them).  So yes, it has an `appengine-web.xml` in it's tree (located at `src/main/webapp/WEB-INF/` from the module root). 

It gets packaged as a WAR, so that `WEB-INF/` ends up at the top-level of the WAR. See below:

![tree](https://user-images.githubusercontent.com/102831/34052819-d9c76a24-e178-11e7-994e-b83b6a80d10e.png)

**Comment by etanshaul** [comment]
@eschultink,

You mentioned that you suddenly lost the target `module-frontend:war exploded`. While I'm not yet sure what caused it to vanish, a quick checklist of things to look for:

- Does this module have the appengine standard facet (you confirmed that it has an `appengine-web.xml` but I'm wondering if perhaps something wiped out your facets).
- Does this module have a corresponding exploded war artifact (note that it must be an exploded war)

These things can be verified as usual under `File -> Project Structure`.

**Comment by eschultink** [comment]
@etanshaul all still Maven modules that should be GAE services still had the facet. And all expected artifacts still there.

Even weirder: Trying to reset things, I manually removed all the GAE facets from the modules, intending to re-add them.  But then I noticed that my Run Configuration, although lacking any artifacts, appears to work normally - despite the warning. So underneath in the `.idea` config is still correct, even GUI doesn't render the state correctly?

<img width="1126" alt="run debug configurations 2017-12-15 11-53-49" src="https://user-images.githubusercontent.com/102831/34058207-f7e913f2-e18e-11e7-81f6-c1a2bc228198.png">


Project Structure state:
  * no GAE facets. Only "Web" facet for each of the Maven modules that are in reality GAE services (these are built with maven-war plugin).
  * all expected artifacts exist

**Comment by lilianaziolek** [comment]
The plot thickens... ;)
At one point when I had the artifact disappear, I closed intellij, zipped my entire project (along with iml files, .idea folder, build directories - literally everything). I thought: next time when it works, I'll pull out that zip, do a diff, maybe I'll find something. (After starting intellij again at that point it was still broken, just to be clear).
So now for some reason it works again, I unzipped the "broken" package in a new folder, opened it in intellij... and it works! I seriously don't know what to think about it all, but just thought I'd share it here. 

Personally, I'm increasingly convinced that it's not a wrong setting of the project/plugin, but some wicked caching or other temporary state somewhere in intellij that sometimes goes wrong, which would explain flickering back-and-forth between working and not, even when absolutely no changes are done in project config.

**Comment by etanshaul** [comment]
@lilianaziolek @eschultink @freddyboucher @xmeng1 

Sorry for the delay and troubles with this. It looks like the disappearing artifacts from the target list in the App Engine local dev server run configuration (where sometimes it would have artifacts, and then suddenly be empty) was due to a bug in the deployment source collection logic.

This specific issue _should_ be fixed by https://github.com/GoogleCloudPlatform/google-cloud-intellij/pull/1782 which is not yet released into the main repo. We will likely do a patch release of the plugin sometime this week.

For those of you that would like to get this immediately to test out, you can pull the [nightly release](https://plugins.jetbrains.com/plugin/8079-google-cloud-tools) version of the plugin (see the `nightly` tab):
- Open up your plugin settings in intellij: `Settings -> Plugins`
- Click `Browse repositories...`
- Click `Manage repositories...`
- Add the nightly release repo url: `https://plugins.jetbrains.com/plugins/nightly/8079` and click ok
- You can now find the latest nightly release of `Google Cloud Tools` from the list and install it. The version will be a timestamp e.g. `v20171218`

**Comment by lilianaziolek** [comment]
Hi @etanshaul 
So there is progress but still does not work 100%... The artifact now indeed shows in dropdown:
![image](https://user-images.githubusercontent.com/271914/34116732-531e01ea-e411-11e7-95a6-873ac523396c.png)
However, as you can see, it claims that there is no appengine facet in the war. If I look at the facets page:
![image](https://user-images.githubusercontent.com/271914/34116766-71cedcc2-e411-11e7-9972-786bd642b4fd.png)
![image](https://user-images.githubusercontent.com/271914/34116797-84af52c2-e411-11e7-8ab2-e3c16b4c6e0d.png)
Any hints?

**Comment by etanshaul** [comment]
@lilianaziolek which module/artifact are you hoping to run locally? If you wish to run the artifact that is now showing up in the dropdown, then you need to check that the facet is indeed applied to that module. Your module screenshot shows the appengine facet on the `appengine` module. 

I suggest to check the other module and try adding the facet (App Engine standard) to it manually if its missing.

**Comment by lilianaziolek** [comment]
@etanshaul Yes, the module is called appengine :) The artifact (gradle exploded war) is showing in dropdown - but, as you can see on screenshot, when I select it plugin says that App Engine facet is not enabled for appengine gradle artifact. Again, on screenshots you should see that the facet IS added on the appengine intellij module. So no idea why it moans it's not.

**Comment by etanshaul** [comment]
@lilianaziolek sorry, I wasn't very clear. What I meant was that the dropdown shows a _different_ module. Specifically the 'onestopbeauty' module. While this may not get you to where you are trying to be, I was suggesting to check to ensure that the facet is applied to this module (not the `appengine` module).

**Comment by lilianaziolek** [comment]
@etanshaul online.onestopbeauty is a group name, not artifact/module name. The module name is definitely appengine (in drop down too) - my group has a couple artifacts.
Group names often start with org or com, so it might have tricked you ;)

**Comment by etanshaul** [comment]
@lilianaziolek sorry for the delay in response. We've made a few changes and if you'd like to try the new nightly release again and see if it helps. We won't be doing a production release until after the holidays. Also, for the sake of limiting the length of this thread, I've opened up another issue where we can track your particular use-case: https://github.com/GoogleCloudPlatform/google-cloud-intellij/issues/1799 so if you don't mind, lets follow up on this on this new issue.
